src/layers/DropOutLayer.cpp

```diff
@@ -0,0 +1,37 @@
+#include "layers/DropOutLayer.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <random>
+
+namespace itlab_2023 {
+
+void DropOutLayer::run(const Tensor &input, Tensor &output) {
+  switch (input.get_type()) {
+    case Type::kInt: {
+      std::vector<int> vec = *input.as<int>();
+      std::random_device dev;
+      std::mt19937 gen(dev());
+      for (size_t i = 0; i < vec.size(); i++) {
+        if (gen() % (101) < static_cast<float>(drop_rate_) * 100) vec[i] = 0;

```

Comment:
This is more conventional and precise usage of mt19937: https://stackoverflow.com/a/21102720/24543008
It is better to avoid these % and * operations for the sake of:
1. performance (extra operations)
2. readability (it is not clear right away what is done here with % 101)
3. Distribution could be different from uniform (not 100% sure about this one)
Numbers in 0 <= x < 1 can be generated using uniform real distribution

Comment is applicable for both integer and float version. Btw, there is clearly the space for reusing the code, so you can reduce duplication of code here

Reply:
fixed